pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -54,20 +54,82 @@ func NewPredictor(client client.Client, scheme *runtime.Scheme, inferenceService
 
 // Reconcile observes the predictor and attempts to drive the status towards the desired state.
 func (p *Predictor) Reconcile(isvc *v1beta1.InferenceService) error {
-	predictor := isvc.Spec.Predictor.GetImplementation()
+	var container *v1.Container
+	var podSpec v1.PodSpec
+
 	annotations := utils.Filter(isvc.Annotations, func(key string) bool {
 		return !utils.Includes(constants.ServiceAnnotationDisallowedList, key)
 	})
-	// KNative does not support INIT containers or mounting, so we add annotations that trigger the
-	// StorageInitializer injector to mutate the underlying deployment to provision model data
-	if sourceURI := predictor.GetStorageUri(); sourceURI != nil {
-		annotations[constants.StorageInitializerSourceUriInternalAnnotationKey] = *sourceURI
-	}
+
 	hasInferenceLogging := addLoggerAnnotations(isvc.Spec.Predictor.Logger, annotations)
 	hasInferenceBatcher := addBatcherAnnotations(isvc.Spec.Predictor.Batcher, annotations)
+
 	// Add agent annotations so mutator will mount model agent to multi-model InferenceService's predictor
 	addAgentAnnotations(isvc, annotations, p.inferenceServiceConfig)
 
+	// Reconcile modelConfig
+	configMapReconciler := modelconfig.NewModelConfigReconciler(p.client, p.scheme)
+	if err := configMapReconciler.Reconcile(isvc); err != nil {
+		return err
+	}
+
+	// If Model is specified, prioritize using that. Otherwise, we will assume a framework object was specified.
+	if isvc.Spec.Predictor.Model != nil {
+		runtimes, err := isvc.Spec.Predictor.Model.GetSupportingRuntimes(p.client, isvc.Namespace)
```

Comment:
Yea, I can add that check to save some API calls. The current way does work as I borrowed it from ModelMesh-Serving. It's just perhaps sub-optimal for cases where a runtime is explicitly specified since it compares with all ServingRuntimes.

Reply:
I _think_ the model-mesh logic only sets the runtime for placement purposes if it's provided. I don't remember if we also validate that the runtime supports the specified model type if both are provided.